cmake/Modules/FindDPCPP.cmake

```diff
@@ -96,6 +97,11 @@ function(add_sycl_to_target)
     "${multi_value_args}"
     ${ARGN}
   )
+  if((${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM") AND
+      (TUNING_TARGET IN_LIST non_intel_target) )
+    target_compile_options(${SB_ADD_SYCL_TARGET} PRIVATE -fno-fast-math)
```

Comment:
I don't think that having a different flag for nightly and release would be an issue, since it is transparent to users and, if they want, they can always add it themself. Moreover, I am not a fan of adding an unnecessary flag that may limit optimization to a compiler and this particular flag is easier to add that to remove while configuring the project build.
If in future nightly this become an issue we can add it later, what do you think?

I am happy to add a message when we add this flag.

Reply:
We agreed it makes sense to add `-fno-fast-math` only when we also add `-mllvm -loopopt=0` i.e. for every target except `INTEL_GPU`.